eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducer.java

```diff
@@ -0,0 +1,313 @@
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.amqp.exception.ErrorCondition;
+import com.azure.core.implementation.util.ImplUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.eventhubs.implementation.AmqpSendLink;
+import com.azure.messaging.eventhubs.implementation.ErrorContextProvider;
+import com.azure.messaging.eventhubs.implementation.EventDataUtil;
+import org.apache.qpid.proton.message.Message;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+/**
+ * A producer responsible for transmitting {@link EventData} to a specific Event Hub, grouped together in batches.
+ * Depending on the options specified at creation, the producer may be created to allow event data to be automatically
+ * routed to an available partition or specific to a partition.
+ *
+ * <p>
+ * Allowing automatic routing of partitions is recommended when:
+ * <ul>
+ * <li>The sending of events needs to be highly available.</li>
+ * <li>The event data should be evenly distributed among all available partitions.</li>
+ * </ul>
+ * </p>
+ *
+ * <p>
+ * If no partition is specified, the following rules are used for automatically selecting one:
+ * <ol>
+ * <li>Distribute the events equally amongst all available partitions using a round-robin approach.</li>
+ * <li>If a partition becomes unavailable, the Event Hubs service will automatically detect it and forward the
+ * message to another available partition.</li>
+ * </ol>
+ * </p>
+ *
+ * @see EventHubClient#createProducer()
+ */
+public class EventHubProducer implements Closeable {
+    /**
+     * The default maximum allowable size, in bytes, for a batch to be sent.
+     */
+    public static final int MAX_MESSAGE_LENGTH_BYTES = 256 * 1024;
+
+    private static final int MAX_PARTITION_KEY_LENGTH = 128;
+    private static final SendOptions DEFAULT_SEND_OPTIONS = new SendOptions();
+
+    private final ClientLogger logger = new ClientLogger(EventHubProducer.class);
+    private final AtomicBoolean isDisposed = new AtomicBoolean();
+    private final EventHubProducerOptions senderOptions;
+    private final Mono<AmqpSendLink> sendLinkMono;
+    private final boolean isPartitionSender;
+
+    /**
+     * Creates a new instance of this {@link EventHubProducer} with batches that are {@code maxMessageSize} and sends
+     * messages to {@code partitionId}.
+     */
```

Comment:
`maxMessageSize` and `partitionId` are not args into this constructor (not that it matters as this isn't public API)

Reply:
Fixed